parsec/backend/asgi/rpc.py

```diff
@@ -75,3 +104,184 @@ async def anonymous_api(raw_organization_id: str):
     # Run command
     rep = await cmd_func(client_ctx, msg)
     return rpc_msgpack_rep(rep)
+
+
+rpc_auth_bp = Blueprint("authenticated_api", __name__)
+
+
+@rpc_auth_bp.route("/authenticated/<raw_organization_id>", methods=["POST"])
+async def authenticated_api(raw_organization_id: str):
+    backend: BackendApp = g.backend
+    logger = current_app.logger.getChild(authenticated_api.__name__)
+
+    logger.debug(
+        "Received call in the authenticated api for the organization {org_id} from the device={dev_id}".format(
+            org_id=raw_organization_id, dev_id=request.headers.get("Author", "Missing")
+        )
+    )
+    try:
+        organization_id = OrganizationID(raw_organization_id)
+    except ValueError:
+        logger.warning(f"Invalid organization id: {raw_organization_id}")
+        abort(404)
+
+    try:
+        organization = await backend.organization.get(organization_id)
+    except OrganizationNotFoundError:
+        logger.warning(f"Organization {organization_id} not found")
+        abort(400)
+    else:
+        if organization.is_expired:
+            logger.warning(f"Organization {organization_id} is expired")
+            abort(400)
+
+    body: bytes = await request.get_data(cache=False)
+    try:
+        user, device = await verify_auth_request(body, request.headers, organization, backend)
+    except (HttpProtocolError, UserNotFoundError) as e:
+        logger.warning(f"Invalid request, cause: {e}")
+        abort(400)
+    except AuthenticationProtocolError as e:
+        logger.warning(f"Failed to authenticate request because of {e}")
+        abort(401)
+
+    api_version = get_api_version(request.headers)
+
+    client_ctx = build_client_context(organization, user, device, api_version)
+
+    # Unpack the verified body
+    if request.headers["Content-Type"] != "application/msgpack":
+        logger.warning("Invalid content-type {}".format(request.headers["Content-Type"]))
+        abort(400)
+    try:
+        msg = unpackb(body)
+    except SerdePackingError:
+        return rpc_msgpack_rep({"status": "invalid_msg_format"})
+
+    cmd = msg.get("cmd")
+    try:
+        if not isinstance(cmd, str):
+            raise KeyError()
+        cmd_func = backend.apis[ClientType.AUTHENTICATED][cmd]
+    except KeyError:
```

Comment:
I'm not sure it makes sense to factorize `if not isinstance(cmd, str):` within this try-except. Maybe perform this check before and have a dedicated warning for this case?

Reply:
`mypy` _require_ this check to validate that `cmd` is indeed a string

Reply:
I've added more context to the raised exception, this would help with the logger just below in the `catch`